subworkflows/local/scaffold/main.nf

```diff
@@ -20,135 +8,33 @@ workflow SCAFFOLD {
+    ch_scaffolded_assemblies = Channel.empty()
+    ch_logs                  = Channel.empty()
 
-    SAMTOOLS_FAIDX.out.fai
-        .map{ meta, fai ->
-            fai.splitCsv( sep: '\t', header: false )
-                .collect{ row -> row[ 0..1 ].join('\t') }
-                .join('\n')
-        }.collectFile()
-        .set { chrom_sizes }
-
-    // Combine hi-c alignment with chrom sizes
-    BWAMEM2_MEM_SCAFFOLD.out.bam.combine(chrom_sizes)
-        .multiMap{ meta, hic_bam, chr_lengths ->
-            hicbams: [ meta, hic_bam ]
-            lengths: chr_lengths
-        }.set{ pairtools_parse_input }
-
-    PAIRTOOLS_PARSE (
-        pairtools_parse_input.hicbams,
-        pairtools_parse_input.lengths
-    )
-    PAIRTOOLS_SORT(PAIRTOOLS_PARSE.out.pairsam)
-    PAIRTOOLS_SORT.out.sorted.groupTuple()
-        .branch { meta, pairsam ->
-            single: pairsam.size() == 1
-                return [ meta, *pairsam ]
-            multi: true
-                return [ meta, pairsam ]
-        }
-        .set { pairtools_sorted }
-    PAIRTOOLS_MERGE(pairtools_sorted.multi)
-    PAIRTOOLS_DEDUP(
-        PAIRTOOLS_MERGE.out.pairs.mix(
-            pairtools_sorted.single
+    if (params.hic_pipeline == 'omni-c')
+    {
+        SCAFFOLD_OMNIC_BWA(
+            ch_assemblies,
+            ch_hic
         )
-    )
-
-    joinByMetaKeys(
-        PAIRTOOLS_DEDUP.out.pairs,
-        getPrimaryAssembly(ch_assemblies),
-        keySet: ['id','sample'],
-        meta: 'rhs'
-    ).multiMap{ meta, pairs, fasta ->
-        pairs: [ meta, pairs ]
-        fasta: [ meta, fasta ]
-    }.set { pairtools_split_input }
-
-    PAIRTOOLS_SPLIT(
-        pairtools_split_input.pairs,
-        pairtools_split_input.fasta,
-        true    // sort_bam
-    )
-
-    combineByMetaKeys( // Combine (bam + assembly) with fai
-        combineByMetaKeys( // Combine bams with assembly
-            PAIRTOOLS_SPLIT.out.bam,
-            getPrimaryAssembly(ch_assemblies),
-            keySet: ['id','sample'],
-            meta: 'rhs'
-        ),
-        SAMTOOLS_FAIDX.out.fai,
-        keySet: ['id','sample'],
-        meta: 'rhs'
-    ).multiMap{ meta, bam, fasta, fai ->
-        bam:   [ meta, bam ]
-        fasta: [ fasta ]
-        fai:   [ fai ]
-    }.set{ yahs_input }
-
-    YAHS(
-        yahs_input.bam,
-        yahs_input.fasta,
-        yahs_input.fai
-    )
-
-    // Consensus case:
-    // Preserve haplotigs from purge dups
-    ch_scaff_and_alt = ch_assemblies.map { meta, assembly -> [ meta, assembly.alt_fasta ] }
-        .mix( YAHS.out.scaffolds_fasta )
-    ch_scaffolded_assemblies = constructAssemblyRecord( ch_scaff_and_alt, false )
-
-    PAIRTOOLS_PARSE.out.stat
-        .mix (
-            PAIRTOOLS_DEDUP.out.stat
+        ch_versions = ch_versions.mix(SCAFFOLD_OMNIC_BWA.out.versions)
+        ch_scaffolded_assemblies = ch_scaffolded_assemblies.mix(SCAFFOLD_OMNIC_BWA.out.assemblies)
+        ch_logs = ch_logs.mix(SCAFFOLD_OMNIC_BWA.out.logs)
```

Comment:
Given the initial channels are `empty()`, you don't need to mix here. You can just assign.

```suggestion
        ch_versions = SCAFFOLD_OMNIC_BWA.out.versions
        ch_scaffolded_assemblies = SCAFFOLD_OMNIC_BWA.out.assemblies
        ch_logs = SCAFFOLD_OMNIC_BWA.out.logs
```

You only need to `mix` if there might be content in the channels beforehand.